src/client/mapview.cpp

```diff
@@ -297,6 +320,8 @@ void MapView::drawMapForeground(const Rect& rect)
             c.first->drawInformation(c.second, g_map.isCovered(c.first->getPrewalkingPosition(), m_cachedFirstVisibleFloor), rect, flags);
         }
     }
+	
+	drawTileWidget(rect, srcRect);
```

Comment:
shouldn't they be draw in drawFloor? Now widgets will cover upper floors

Reply:
Can't be there, due to floor Rect scaling. Widgets end up rendered too big and on wrong position. It's not an issue that they cover upper floors. These widgets are drawn only on players current floor, so even if they are ex. on top of a roof, players will be aware that they are still on their floor.